share/completions/port.fish

```diff
@@ -0,0 +1,115 @@
+#completion for port
+
+function __fish_port_no_subcommand --description 'Test if port has yet to be given the subcommand'
+	for i in (commandline -opc)
+		if contains -- $i activate archive build cat cd checksum clean configure \
+				contents deactivate dependents deps destroot dir distcheck dmg echo \
+				edit extract fetch file gohome info install installed lint list \
+				livecheck location load log logfile mirror mdmg mpkg notes outdated \
+				patch pkg provides rdependents rdeps reload rev search select \
+				selfupdate setrequested setunrequested sync test unarchive uninstall \
+				unload unsetrequested upgrade url usage variants version work
+			return 1
+		end
+	end
+	return 0
+end
+
+function __fish_port_use_package --description 'Test if port command should have packages as potential completion'
+	for i in (commandline -opc)
+	if contains -- $i activate archive build cat cd checksum clean configure \
+			contents deactivate dependents deps destroot dir distcheck dmg echo \
+			edit extract fetch file gohome info install installed lint list \
+			livecheck location load log logfile mirror mdmg mpkg notes outdated \
+			patch pkg provides rdependents rdeps reload rev search select \
+			selfupdate setrequested setunrequested sync test unarchive uninstall \
+			unload unsetrequested upgrade url usage variants version work
+		return 0
+		end
+	end
+	return 1
+end
+
+complete -c port -n '__fish_port_use_package' -a '(__fish_print_packages)' --description 'Package'
+
+complete -f -n '__fish_port_no_subcommand' -c port -a 'activate'  --description 'Set the status of an previously installed version of a port to active'
```

Comment:
I'm not sure if the description text here comes from the manual page, but it might be better to use shorter and more active voiced descriptions: eg "Activate a previously-installed port". This comment applies to many of the descriptions below - more than 50 or so characters will generally be truncated on smaller screens.

Reply:
Yes, those are taken from the respective manpages. I'll try and shorten the more verbose descriptions.